app/src/main/java/Bamboo/view/MainFrame.java

```diff
@@ -50,6 +48,35 @@ private void buildFrame()
         setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
         setResizable(false);
         setLayout(new BorderLayout());
+        setJMenuBar(new MenuBar(this));
+    }
+
+
+    public void quit() {
+        Object[] options = {"Yes","No"};
+
+        int n = JOptionPane.showOptionDialog(this, "Do you want to save and quit? ",
+                "Bamboo", JOptionPane.YES_NO_CANCEL_OPTION, JOptionPane.WARNING_MESSAGE, null, options,
+                options[1]);
+
+        if (n == JOptionPane.NO_OPTION) {
+            System.exit(0);
+        }
+        else {
+            saveAndQuit();
+        }
```

Comment:
This if statement (Since Java v8) doesn't need to have braces as they are all just one line statements, so you can rewrite like this:
```
        if (n == JOptionPane.NO_OPTION)
            System.exit(0);
        else
            saveAndQuit();
```
Much neater and easier to read!